### PR TITLE
DBALConnectionFactory opens db connection, even if its not necessary

### PR DESCRIPTION
**Pull request by IceShack**
Hi guys,

We sometimes got the issue that our app can't connect to our database.
Right now we just get a "ServiceNotCreatedException", because the DBALConnectionFactory opens a db connection while creating the connection services.

This fix only opens a db connection, if needed.
That way we can catch the ConnectionException in our controller/business logic and return a proper response.

### CODE REVIEW

File: src/DoctrineORMModule/Service/DBALConnectionFactory.php [hunk 1]
```diff
@@ -60,13 +60,16 @@ public function __invoke(ContainerInterface $container, $requestedName, array $o
...
 
-        foreach ($options->getDoctrineCommentedTypes() as $type) {
-            $platform->markDoctrineTypeCommented(Type::getType($type));
+        if (count($options->getDoctrineTypeMappings()) > 0 || $options->getDoctrineCommentedTypes() > 0) {
+            $platform = $connection->getDatabasePlatform();
+            foreach ($options->getDoctrineTypeMappings() as $dbType => $doctrineType) {
+                $platform->registerDoctrineTypeMapping($dbType, $doctrineType);
+            }
+
+            foreach ($options->getDoctrineCommentedTypes() as $type) {
```
Comment (Ocramius) [inline src/DoctrineORMModule/Service/DBALConnectionFactory.php]: This is a hidden addition that isn't tested

File: src/DoctrineORMModule/Service/DBALConnectionFactory.php [hunk 2]
```diff
@@ -60,13 +60,16 @@ public function __invoke(ContainerInterface $container, $requestedName, array $o
         $eventManager  = $container->get($options->getEventManager());
 
         $connection = DriverManager::getConnection($params, $configuration, $eventManager);
-        $platform = $connection->getDatabasePlatform();
-        foreach ($options->getDoctrineTypeMappings() as $dbType => $doctrineType) {
-            $platform->registerDoctrineTypeMapping($dbType, $doctrineType);
```
Comment (Ocramius) [inline src/DoctrineORMModule/Service/DBALConnectionFactory.php]: A simpler fix is to simply this change this line to `$connection->getDatabasePlatform()->registerDoctrineTypeMapping($dbType, $doctrineType);`

File: src/DoctrineORMModule/Service/DBALConnectionFactory.php (hunk 1 above)
Comment (IceShack) [inline src/DoctrineORMModule/Service/DBALConnectionFactory.php]: That was there all along, i only added the if.

**Comment by Ocramius** [comment]
LGTM 👍

**Pull request merged**